smartsim/_core/_install/builder.py

```diff
@@ -578,3 +643,231 @@ def _move_torch_libs(self) -> None:
         if sys.platform == "darwin":
             dylibs = pip_torch_path / ".dylibs"
             self.copy_dir(dylibs, ss_rai_torch_path / ".dylibs", set_exe=True)
+
+
+def _threaded_map(fn: t.Callable[[_T], _U], items: t.Iterable[_T]) -> t.Sequence[_U]:
+    items = tuple(items)
+    if not items:  # No items so no work to do
+        return ()
+    num_workers = min(len(items), (os.cpu_count() or 8) * 5)
```

Comment:
The default for [ThreadPoolExecutor](`https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor`) is the CPU count times 5 based on the assumptions that:

a) threads are more light-weight than processes
b) the thread pool is being used for (relatively) short lived tasks
c) the work is IO bound

I figured these seemed like reasonable assumptions to make that would be good "ceiling number" on the unlikely chance someone throws a lot of items to map over.

Totally fair to drop the default CPU count to 4 though! Will add!!

Reply:
Huh, that's interesting. I get it but it just seems like a completely arbitrary number. Fine by me especially since it follows the convention, but yeah drop it to 4.